src/classlibnative/bcltype/bignum.cpp

```diff
@@ -0,0 +1,597 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: bignum.cpp
+//
+
+//
+
+#include "bignum.h"
+
+UINT32 BigNum::m_power10UInt32Table[UINT32POWER10NUM] =
+{
+    1,          // 10^0
+    10,         // 10^1
+    100,        // 10^2
+    1000,       // 10^3
+    10000,      // 10^4
+    100000,     // 10^5
+    1000000,    // 10^6
+    10000000,   // 10^7
+};
+
+BigNum BigNum::m_power10BigNumTable[BIGPOWER10NUM];
+BigNum::StaticInitializer BigNum::m_initializer;
+
+BigNum::BigNum()
+    :m_len(0)
+{
+}
+
+BigNum::BigNum(UINT32 value)
+{
+    SetUInt32(value);
+}
+
+BigNum::BigNum(UINT64 value)
+{
+    SetUInt64(value);
+}
+
+BigNum::~BigNum()
+{
+}
+
+BigNum& BigNum::operator=(const BigNum &rhs)
+{
+    UINT8 length = rhs.m_len;
+    UINT32* pCurrent = m_blocks;
+    const UINT32* pRhsCurrent = rhs.m_blocks;
+    const UINT32* pRhsEnd = pRhsCurrent + length;
+
+    while (pRhsCurrent != pRhsEnd)
+    {
+        *pCurrent = *pRhsCurrent;
+
+        ++pCurrent;
+        ++pRhsCurrent;
+    }
+
+    m_len = length;
+
+    return *this;
+}
+
+int BigNum::Compare(const BigNum& lhs, UINT32 value)
```

Comment:
I can overload these operators but I think we should do this enhancement in another PR because a simple compare method should be enough to implement the algorithm.